client/src/main/java/org/craftercms/profile/services/impl/ProfileServiceRestClient.java

```diff
@@ -298,15 +298,7 @@ public Profile getProfileByTicket(String ticketId, String... attributesToReturn)
         String url = getAbsoluteUrl(BASE_URL_PROFILE + URL_PROFILE_GET_BY_TICKET);
         url = addQueryParams(url, params, false);
 
-        try {
-            return doGetForObject(url, Profile.class);
-        } catch (ProfileRestServiceException e) {
-            if (e.getStatus() == HttpStatus.NOT_FOUND) {
-                return null;
-            } else {
-                throw e;
-            }
-        }
+        return doGetForObject(url, Profile.class);
```

Comment:
I don't think you have to get rid of the try catch, because we still want to return null when there's no profile. Instead, if status is NOT FOUND, and error code is NO SUCH TICKET, throw the exception.